src/circuit.rs

```diff
@@ -92,6 +92,91 @@ pub struct Cell {
     column: Column<Any>,
 }
 
+/// An assigned cell.
+#[derive(Clone, Copy, Debug)]
+pub struct AssignedCell<F: Field, T: Copy + Into<F>> {
+    value: Option<T>,
+    cell: Cell,
+    _marker: PhantomData<F>,
+}
+
+impl<F: Field, T: Copy + Into<F>> AssignedCell<F, T> {
+    /// Assign a new AssignedCell.
+    pub fn assign<A, AR>(
+        region: &mut Region<'_, F>,
+        annotation: A,
+        column: Column<Any>,
+        offset: usize,
+        value: Option<T>,
+    ) -> Result<Self, Error>
+    where
+        A: Fn() -> AR,
+        AR: Into<String>,
+    {
+        let cell = {
+            let value: Option<F> = value.map(|v| v.into());
+            match column.column_type() {
+                Any::Advice => {
+                    region.assign_advice(annotation, column.try_into().unwrap(), offset, || {
+                        value.ok_or(Error::SynthesisError)
+                    })?
+                }
+                Any::Fixed => {
+                    region.assign_fixed(annotation, column.try_into().unwrap(), offset, || {
+                        value.ok_or(Error::SynthesisError)
+                    })?
+                }
+                _ => unreachable!(),
```

Comment:
This should be a `panic!("cells in an instance column cannot be assigned to")` or an error, and the documentation for the method should explain.